plugins/inputs/execd/execd.go

```diff
@@ -0,0 +1,186 @@
+package execd
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"log"
+	"os/exec"
+	"sync"
+	"time"
+
+	"github.com/kballard/go-shellquote"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/influxdata/telegraf/plugins/parsers"
+)
+
+const sampleConfig = `
+  ## Program to run as daemon
+  command = "telegraf-smartctl"
+
+  ## Define how the process is signaled on each collection interval.
+
+  ## Valid values are:
+  ##   "none"   : Do not signal anything.
+  ##              The process must output metrics by itself.
+  ##   "STDIN"  : Send a newline on STDIN.
+  ##   "SIGHUP" : Send a HUP signal. Not available on Windows.
+  signal = "none"
+
+  ## Data format to consume.
+  ## Each data format has its own unique set of configuration options, read
+  ## more about them here:
+  ## https://github.com/influxdata/telegraf/blob/master/docs/DATA_FORMATS_INPUT.md
+  data_format = "influx"
+`
+
+type Execd struct {
+	Command string
+	Signal  string
+
+	acc     telegraf.Accumulator
+	cmd     *exec.Cmd
+	parser  parsers.Parser
+	stdin   io.WriteCloser
+	stopped bool
+}
+
+func (e *Execd) SampleConfig() string {
+	return sampleConfig
+}
+
+func (e *Execd) Description() string {
+	return "Run executable as long-running input plugin"
+}
+
+func (e *Execd) SetParser(parser parsers.Parser) {
+	e.parser = parser
+}
+
+func (e *Execd) Start(acc telegraf.Accumulator) error {
+	e.acc = acc
+
+	args, err := shellquote.Split(e.Command)
+	if err != nil || len(args) == 0 {
+		return fmt.Errorf("unable to parse command: %s", err)
+	}
+
+	go e.cmdRun(args)
+
+	return nil
+}
+
+func (e *Execd) Stop() {
+	e.stopped = true
```

Comment:
This would be obsolete when starting command in `Gather`.

Should there be a signal sent to `e.cmd.Process` to tell the command to stop? If so, should it be configurable?

Reply:
It would probably be a good idea. Very nice would be to send TERM, wait for shutdown for a responsable amount of time, and finally kill with KILL. This would give the process some time to e.g. cleanup things.

I'm also just reading into process groups and `cmd.SysProcAttr` for Go, but I have absolutely no idea about Windows.

Reply:
Changed in 0e67fe56.